grammars/erlang.bnf

```diff
@@ -321,6 +324,32 @@ argument_definition_list ::= '(' argument_definition? (',' argument_definition)*
 clause_guard ::= when <<guarded guard>> {pin=1}
 clause_body ::= '->' exprs {pin=1}
 
+app_expression ::= empty '{' 'application' ',' module_ref ',' app_parameters '}' {elementType=tuple_expression}
+app_parameters ::= '[' app_parameter? (',' app_parameter)* ']' {elementType=list_expression}
+app_parameter ::= '{'
+   ('description' ',' string_literal
+  | 'id' ',' string_literal
+  | 'vsn' ',' string_literal
+  | 'modules' ',' app_modules
+  | 'maxP' ',' integer
+  | 'maxT' ',' integer
+  | 'registered' ',' app_modules
+  | 'included_applications' ',' app_applications
+  | 'applications' ',' app_applications
+  | 'env' ',' app_env
+  | 'mod' ',' app_mod
+  | 'start_phases' ',' expression
+  | 'runtime_dependencies' ',' app_runtime_deps
+  | config_exprs)
+  '}' {methods=[getName]}
```

Comment:
No. I've introduced this PSI element for resolving of modules. Name of commit and name of pull request alludes to it.
I think it's the most natural solution. When we type `-behaviour(<caret>)`, we already know that there will be module. This scenario is identical. When we are in app file and typing `modules, [<caret>]`, we already know that there will be list of modules.

I didn't want to change ImportedOtpApp at all, but it depended on old PSI structure. So I've changed PSI structure and refresh ImportedOtpApp.